components/wpt-results.html

```diff
@@ -46,59 +120,192 @@
 
       static get properties() {
         return {
+          query: {
+            type: String,
+            value: '',
+            observer: '_queryChanged'
+          },
           specs: {
             type: Array,
-            observer: '_specsChanged',
             value: []
-          }
+          },
+          testRuns: {
+            type: Array,
+          },
         };
       }
 
-      connectedCallback() {
+      async connectedCallback() {
         super.connectedCallback();
 
-        this._fetchResults()
+        const testFileResults = await Promise.all(this.testRuns.map(testRun => {
+          return this._fetchResults(testRun.results_url)
+        }))
```

Comment:
tl;dr: `Promise.all` waits for all `_fetchResults` promises to return before continuing on. `testFileResults` is an array of results from each `_fetchResults` call.